nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -476,6 +476,17 @@ def __call__(self, x: List[Tensor]):
         return self._reduce_out_of_place(x)
 
 
+class ShapeReducer(TensorReducerBase):
+    def __init__(self):
+        super().__init__(inplace=False)
+
+    def _reduce_out_of_place(self, x: List[TensorType]) -> List[TensorType]:
+        return [Tensor(x[0].shape)]
+
+    def get_inplace_fn(self) -> Optional[InplaceInsertionFNType]:
+        return None
```

Comment:
Inplace OV  implementation could possibly reduce memory consumption even further, can we try it out?

Reply:
Done, don't see any significant improvements though